go/libraries/doltcore/sqle/expeval/expression_evaluator.go

```diff
@@ -0,0 +1,254 @@
+			return nil, err
+		}
+
+		rightFunc, err := getExpFunc(nbf, sch, typedExpr.Right)
+
+		if err != nil {
+			return nil, err
+		}
+
+		return newAndFunc(leftFunc, rightFunc), nil
+	}
+
+	return nil, errNotImplemented.New(exp.Type().String())
+}
+
+func newOrFunc(left ExpressionFunc, right ExpressionFunc) ExpressionFunc {
+	return func(ctx context.Context, vals map[uint64]types.Value) (b bool, err error) {
+		lRes, err := left(ctx, vals)
+
+		if err != nil {
+			return false, err
+		}
+
+		if lRes {
+			return true, nil
+		}
+
+		return right(ctx, vals)
+	}
+}
+
+func newAndFunc(left ExpressionFunc, right ExpressionFunc) ExpressionFunc {
+	return func(ctx context.Context, vals map[uint64]types.Value) (b bool, err error) {
+		lRes, err := left(ctx, vals)
+
+		if err != nil {
+			return false, err
+		}
+
+		if !lRes {
+			return false, nil
+		}
+
+		return right(ctx, vals)
+	}
+}
+
+type comparisonType int
+
+const (
+	invalidCompare comparisonType = iota
+	variableConstCompare
+	variableVariableCompare
+	constConstCompare
+)
+
+func getComparisonType(be expression.BinaryExpression) ([]*expression.GetField, []*expression.Literal, comparisonType, error) {
+	var variables []*expression.GetField
+	var consts []*expression.Literal
+
+	for _, curr := range []sql.Expression{be.Left, be.Right} {
+		// need to remove this and handle properly
+		if conv, ok := curr.(*expression.Convert); ok {
+			curr = conv.Child
+		}
+
+		switch v := curr.(type) {
+		case *expression.GetField:
+			variables = append(variables, v)
+		case *expression.Literal:
+			consts = append(consts, v)
+		default:
+			return nil, nil, invalidCompare, errUnsupportedComparisonType.New()
+		}
+	}
+
+	var compType comparisonType
+	if len(consts) == 2 {
+		compType = constConstCompare
+	} else if len(variables) == 2 {
+		compType = variableVariableCompare
+	} else {
+		compType = variableConstCompare
+	}
+
+	return variables, consts, compType, nil
+}
+
+var trueFunc = func(ctx context.Context, vals map[uint64]types.Value) (b bool, err error) { return true, nil }
+var falseFunc = func(ctx context.Context, vals map[uint64]types.Value) (b bool, err error) { return false, nil }
+
+func newCamparisonFunc(op compareOp, exp expression.BinaryExpression, sch schema.Schema) (ExpressionFunc, error) {
+	vars, consts, compType, err := getComparisonType(exp)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if compType == constConstCompare {
+		res, err := op.compareLiterals(consts[0], consts[1])
+
+		if err != nil {
+			return nil, err
+		}
+
+		if res {
+			return trueFunc, nil
+		} else {
+			return falseFunc, nil
+		}
+	} else if compType == variableConstCompare {
+		colName := vars[0].Name()
+		col, ok := sch.GetAllCols().GetByNameCaseInsensitive(colName)
+
+		if !ok {
+			return nil, errUnknownColumn.New(colName)
+		}
+
+		tag := col.Tag
+		nomsVal, err := literalToNomsValue(col.Kind, consts[0])
+
+		if err != nil {
+			return nil, err
+		}
+
+		compareNomsValues := op.compareNomsValues
```

Comment:
Why use this variable instead of just using op in the closure?

Reply:
I actually ran benchmarks, and this is faster.